rmgpy/data/kinetics/family.py

```diff
@@ -1438,14 +1449,17 @@ def applyRecipe(self, reactantStructures, forward=True, unique=True):
 
         if not forward:
             template = self.reverseTemplate
+            productNum = self.reactantNum or len(template.products)
```

Comment:
It actually seems to be correct, because `template` has just become the reverse template so `template.products` is the length of the actual products (of the reversed reaction).